support/types.go

```diff
@@ -0,0 +1,120 @@
+	// file and then setup the cluster accordingly. This can be used to provide input such as kind config
+	CreateWithConfig(ctx context.Context, configFile string) (string, error)
+
+	// GetKubeconfig provides a way to extract the kubeconfig file associated with the cluster in question
+	// using the cluster provider native way
+	GetKubeconfig() string
+
+	// GetKubectlContext is used to extract the kubectl context to be used while performing the operation
+	GetKubectlContext() string
+
+	// ExportLogs is used to export the cluster logs via the cluster provider native workflow. This
+	// can be used to export logs from the cluster after test failures for example to analyze the test
+	// failures better after the fact.
+	ExportLogs(ctx context.Context, dest string) error
+
+	// Destroy is used to cleanup a cluster brought up as part of the test workflow
+	Destroy(ctx context.Context) error
+
+	// LoadImage is used to load a set of Docker images to the cluster via the cluster provider native workflow
+	// Not every provider will have a mechanism like this/need to do this. So, providers that do not have this support
+	// can just provide a no-op implementation to be compliant with the interface
+	LoadImage(ctx context.Context, image string) error
+
+	// LoadImageArchive is used to provide a mechanism where a tar.gz archive containing the docker images used
+	// by the services running on the cluster can be imported and loaded into the cluster prior to the execution of
+	// test if required.
+	// Not every provider will have a mechanism like this/need to do this. So, providers that do not have this support
+	// can just provide a no-op implementation to be compliant with the interface
+	LoadImageArchive(ctx context.Context, archivePath string) error
+
+	// SetDefaults is a handler function invoked after creating an object of type E2EClusterProvider. This method is
+	// invoked as the first step after creating an object in order to make sure the default values for required
+	// attributes are setup accordingly if any.
+	SetDefaults() E2EClusterProvider
+
+	// WaitForControlPlane is a helper function that can be used to indiate the Provider based cluster create workflow
+	// that the control plane is fully up and running. This method is invoked after the Create/CreateWithConfig handlers
+	// and is expected to return an error if the control plane doesn't stabilize. If the provider being implemented
+	// does not have a clear mechanism to identify the Control plane readiness or is not required to wait for the control
+	// plane to be ready, such providers can simply add a no-op workflow for this function call.
+	// Returning an error message from this handler will stop the workflow of e2e-framework as returning an error from this
+	// is considered as  failure to provision a cluster
+	WaitForControlPlane(ctx context.Context, client klient.Client) error
+}
+
+type E2EClusterProviderWithK8sHelper interface {
```

Comment:
Hi @harshanarayana 
I am not sure this separate type is needed, just to return Kubneretes clients.
Would having Type `E2EClusterProvider` return an env.Config (which has the client) be enough, as it's done in other part of the code?  If not, would can we return something from which the client can be built rest.Config or something?

Reply:
I am fine to return just the `rest.Config`. Which was my original plan too. https://github.com/kubernetes-sigs/e2e-framework/pull/246#discussion_r1252678904

I thought having a way to easily access this might help some use cases, but then again, the advantages are not too many. I will just add a function that can return `rest.Config` so that it can be used accordingly for any needs.